gpu-alloc/src/linear.rs

```diff
@@ -198,18 +192,19 @@ where
                     let chunk = self.ready.as_mut().expect(
                         "Chunk index is out of bounds. Probably incorrect allocator instance",
                     );
-                    chunk.allocated -= 1;
-                    if chunk.allocated == 0 {
+                    drop(block);
+
+                    if is_unique(&mut chunk.memory) {
```

Comment:
you could also do `Arc::get_mut` here and scrap `is_unique` completely

Reply:
`Arc::get_mut` costs more

Reply:
Interesting indeed. It does the atomics on the weak count, which you don't want to, reasonably.
You'd need to make it so the `debug_assert` code inside `is_unique` doesn't run in optimized builds. Otherwise, as the PR stands, it also calls `get_mut` internally, anyway.

Reply:
Well, `debug_assert` code does not run in optimized builds by definition of `debug_assert`.

Reply:
Oh it runs, it just doesn't panic. That's one of the weird things I stumbled upon back in the day.

Reply:
Nope. It doesn't run, but compiles.
`debug_assert!(expr)` expands to `if cfg!(debug_assertions) { assert!(expr); }`

Reply:
hmm, indeed! I got confused here

Reply:
I was thinking about the type checks (i.e. using something that's under `#[cfg(debug_assertions)]` is not possible in there), not run-time, so that was my confusion. Thank you for correcting me!